### ISSUE TITLE
Add support to provide an external ObjectMapper

### ISSUE THREAD

**Issue by srinivasvsk** [vladmihalcea/hibernate-types#3]
Firstly,   greatly appreciate your work on simplifying the json configuration for hibernate. 
While using this package with kotlin though, ObjectMapper needs some additional configuration,  would you look at externalizing the objectmapper, so it can be configured from outside or re-used.

Thanks again

**Comment by vladmihalcea** [comment]
Just send me a Pull Request with your proposed changes and I'll review them and let you know if they need any further improvements.

**Comment by UltimaPhoenix** [comment]
I think that this is an important feature, I will send you a Pull Request for this.

**Comment by vladmihalcea** [comment]
@UltimaPhoenix There already [this branch](https://github.com/vladmihalcea/hibernate-types/tree/CustomObjectMapper) where I started the development. I just didn't have time to finish it for all modules. 

You can try it. It works just fine on the branch.

**Comment by mguilherme** [comment]
Are there any news regarding this issue @vladmihalcea ?

Right now I'm providing additional configuration to the ObjectMapper like this:

```Java
	/**
	 * {@link JacksonUtil#OBJECT_MAPPER} configuration.
	 */
	static {
		JacksonUtil.OBJECT_MAPPER.configure( DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false );

//		JacksonUtil.OBJECT_MAPPER.configure( SerializationFeature.INDENT_OUTPUT, true );
		JacksonUtil.OBJECT_MAPPER.configure( SerializationFeature.FAIL_ON_EMPTY_BEANS, false );

		JacksonUtil.OBJECT_MAPPER.enableDefaultTyping( ObjectMapper.DefaultTyping.NON_CONCRETE_AND_ARRAYS, As.PROPERTY );
		JacksonUtil.OBJECT_MAPPER.disable( SerializationFeature.WRITE_DATES_AS_TIMESTAMPS );

		JacksonUtil.OBJECT_MAPPER.registerModules( new JavaTimeModule() );
	}
```

It looks like the `JacksonUtil.OBJECT_MAPPER.enableDefaultTyping( ObjectMapper.DefaultTyping.NON_CONCRETE_AND_ARRAYS, As.PROPERTY );` is not working, I guess If I try to use non concrete types it will fail.

**Comment by vladmihalcea** [comment]
@mguilherme I created a branch as indicated above, but I didn't have time to finish it yet. If you have time, take a look at it, and you can push the development further and send me a Pull Request when you are done. Thanks.

**Comment by mguilherme** [comment]
Thank you @vladmihalcea, will try to give a look.

**Comment by vladmihalcea** [comment]
Implemented and released in 2.1.0.

**Comment by vladmihalcea** [comment]
Here's an article which describes this [new feature](https://vladmihalcea.com/hibernate-types-customize-jackson-objectmapper/).
